superset/assets/src/explore/components/controls/ColorSchemeControl.jsx

```diff
@@ -47,23 +46,27 @@ export default class ColorSchemeControl extends React.PureComponent {
   }
 
   renderOption(key) {
-    const { schemes } = this.props;
+    const { isLinear, schemes } = this.props;
     const schemeLookup = isFunction(schemes) ? schemes() : schemes;
-    const currentScheme = schemeLookup[key.value || defaultProps.value].colors;
+    const currentScheme = schemeLookup[key.value || defaultProps.value];
 
-    let colors = currentScheme;
-    if (this.props.isLinear) {
-      const colorScaler = colorScalerFactory(currentScheme);
-      colors = [...Array(20).keys()].map(d => (colorScaler(d / 20)));
-    }
+    const colors = isLinear
+      ? currentScheme.getColors(9)
```

Comment:
Why `9`? 🤔 Could perhaps add a comment here.

Reply:
Umm, good point. It should be `10`. 
- `20` is too many
- Maximum colors in our palettes is `10` (I thought it was 9. Just double-check)